mne/decoding/ssd.py

```diff
@@ -0,0 +1,288 @@
+        """Estimate epochs sources given the SSD filters.
+
+        Parameters
+        ----------
+        X : array, shape ([n_epochs, ]n_channels, n_times)
+            The input data from which to estimate the SSD. Either 2D array
+            obtained from continuous data or 3D array obtained from epoched
+            data.
+        y : None | array, shape (n_samples,)
+            Used for scikit-learn compatibility.
+
+        Returns
+        -------
+        X_ssd : array, shape ([n_epochs, ]n_components, n_times)
+            The processed data.
+        """
+        self._check_X(X)
+        if self.filters_ is None:
+            raise RuntimeError('No filters available. Please first call fit')
+
+        X_ssd = self.filters_.T @ X[..., self.picks_, :]
+        # We assume that ordering by spectral ratio is more important
+        # than the initial ordering. This is why we apply component picks
+        # after ordering.
+        sorter_spec = Ellipsis
+        if self.sort_by_spectral_ratio:
+            _, sorter_spec = self.get_spectral_ratio(ssd_sources=X_ssd)
+        if X.ndim == 2:
+            X_ssd = X_ssd[sorter_spec][:self.n_components]
+        else:
+            X_ssd = X_ssd[:, sorter_spec, :][:, :self.n_components, :]
+        return X_ssd
+
+    def get_spectral_ratio(self, ssd_sources):
+        """Get the spectal signal-to-noise ratio for each spatial filter.
+
+        Spectral ratio measure for best n_components selection
+        See :footcite:`NikulinEtAl2011`, Eq. (24).
+
+        Parameters
+        ----------
+        ssd_sources : array
+            Data proyected on source space.
+
+        Returns
+        -------
+        spec_ratio : array, shape (n_channels)
+            Array with the sprectal ratio value for each component.
+        sorter_spec : array, shape (n_channels)
+            Array of indices for sorting spec_ratio.
+
+        References
+        ----------
+        .. footbibliography::
+        """
+        psd, freqs = psd_array_welch(
+            ssd_sources, sfreq=self.info['sfreq'], n_fft=self.n_fft)
+        sig_idx = _time_mask(freqs, *self.freqs_signal)
+        noise_idx = _time_mask(freqs, *self.freqs_noise)
+        if psd.ndim == 3:
+            mean_sig = psd[:, :, sig_idx].mean(axis=2).mean(axis=0)
+            mean_noise = psd[:, :, noise_idx].mean(axis=2).mean(axis=0)
+            spec_ratio = mean_sig / mean_noise
+        else:
+            mean_sig = psd[:, sig_idx].mean(axis=1)
+            mean_noise = psd[:, noise_idx].mean(axis=1)
+            spec_ratio = mean_sig / mean_noise
+        sorter_spec = spec_ratio.argsort()[::-1]
+        return spec_ratio, sorter_spec
+
+    def apply(self):
+        """Not implemented, see ssd.inverse_transform() instead."""
+        # Exists because of _XdawnTransformer
+        raise NotImplementedError('See ssd.inverse_transform()')
+
+    def inverse_transform(self, X):
+        """Remove selected components from the signal.
```

Comment:
this is not an inverse transform

if you follow the API of ICA apply will denoise the data and transform will bring you into "source space"

this deviates from our CSP API